Dockerfile

```diff
@@ -1,5 +1,10 @@
 # Build the manager binary
-FROM golang:1.23 as builder
+ARG GOLANG_VERSION=1.23
+
+FROM registry.access.redhat.com/ubi8/go-toolset:${GOLANG_VERSION} as builder
+ARG TARGETOS
+ARG TARGETARCH
+ARG CGO_ENABLED
```

Comment:
Should we stay consistent with `GOLANG_VERSION`and do:

```
ARG TARGETOS=linux
ARG TARGETARCH
ARG CGO_ENABLED=0
``` 

Instead of the bash expansion below?